src/SDK/Logs/ReadableLogRecord.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\SDK\Logs;
+
+use OpenTelemetry\API\Logs\LogRecord;
+use OpenTelemetry\Context\ContextInterface;
+use OpenTelemetry\SDK\Common\Attribute\AttributesInterface;
+use OpenTelemetry\SDK\Common\Instrumentation\InstrumentationScopeInterface;
+use OpenTelemetry\SDK\Resource\ResourceInfo;
+
+/**
+ * @see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#log-and-event-record-definition
+ * "Note: Typically this will be implemented with a new interface or (immutable) value type."
+ */
+class ReadableLogRecord extends LogRecord
+{
+    private InstrumentationScopeInterface $scope;
+    private LoggerSharedState $loggerSharedState;
+
+    public function __construct(InstrumentationScopeInterface $scope, LoggerSharedState $loggerSharedState, LogRecord $logRecord)
+    {
+        $this->scope = $scope;
+        $this->loggerSharedState = $loggerSharedState;
+
+        parent::__construct($logRecord->body);
+        $this->timestamp = $logRecord->timestamp;
+        $this->observedTimestamp = $logRecord->observedTimestamp;
+        $this->context = $logRecord->context;
+        $this->severityNumber = $logRecord->severityNumber;
+        $this->severityText = $logRecord->severityText;
+        $this->attributes = $logRecord->attributes;
+    }
+
+    public function getInstrumentationScope(): InstrumentationScopeInterface
+    {
+        return $this->scope;
+    }
+
+    public function getResource(): ResourceInfo
+    {
+        return $this->loggerSharedState->getResource();
+    }
+
+    public function getTimestamp(): ?int
+    {
+        return $this->timestamp;
+    }
+
+    public function getObservedTimestamp(): ?int
+    {
+        return $this->observedTimestamp;
+    }
+
+    public function getContext(): ?ContextInterface
+    {
+        return $this->context;
+    }
+
+    public function getSeverityNumber(): ?int
+    {
+        return $this->severityNumber;
+    }
+
+    public function getSeverityText(): ?string
+    {
+        return $this->severityText;
+    }
+
+    /**
+     * @return mixed|null
+     */
+    public function getBody()
+    {
+        return $this->body;
+    }
+
+    public function getAttributes(): AttributesInterface
+    {
+        $factory = $this->loggerSharedState->getLogRecordLimits()->getAttributeFactory();
+        $builder = $factory->builder($this->attributes);
+
+        return $builder->build();
+    }
```

Comment:
We should convert the attributes when constructing the `LogRecord`, otherwise attribute limits won't affect memory usage of records that are queued in a batch processor.

Reply:
I don't think we can do it during LogRecord constructor; LogRecord is an [API method](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/bridge-api.md#logrecord) but attribute limits are an [SDK feature](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/sdk.md#logrecord-limits)

However, I can drop/convert attributes when creating a `ReadableLogRecord` from a `LogRecord` which I think achieves the same thing.

Reply:
We could change `Logger::logRecord(LogRecord)` to `Logger::logRecordBuilder(): LogRecordBuilder` to apply attribute limits on `::setAttribute()`. (I'm not sure how useful that would be considering that the logging implementation will most likely keep references to the original values until the log record is emitted.)

([Java uses Logger::logRecordBuilder()](https://github.com/open-telemetry/opentelemetry-java/blob/ff79d892a379f4b26df22447863616f083dcfcf1/api/logs/src/main/java/io/opentelemetry/api/logs/Logger.java#L29))

Reply:
@Nevay I don't see advantages (at least as I'm writing this) introducing `LogRecordBuilder`. Seems like implementation detail